plugins/modules/zos_lineinfile.py

```diff
@@ -500,32 +573,44 @@ def main():
         if regexp is None and line is None:
             module.fail_json(msg='one of line or regexp is required with state=absent')
 
+    gds = False
+    has_special_chars = False
+    dmod_exec = False
+    return_content = ""
+
     # analysis the file type
+    if "/" not in src:
+        dataset = data_set.MVSDataSet(
+            name=src
+        )
+        src = dataset.name
+        gds = dataset.is_gds_active
+
+    if data_set.DataSet.is_gds_relative_name(src) and gds is False:
+        module.fail_json(msg="{0} does not exist (1)".format(src))
```

Comment:
Here I just have the same question about the error scenario, `gds` variable does look like is used in other places.

Reply:
Is a validation variable to call or no to call deed execution not lineinfile python API